### ISSUE TITLE
[MNG-6704] MavenRepositorySystemUtils.newSession()  misses assignment

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#8543]
**[guofei](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ief2009)** opened **[MNG-6704](https://issues.apache.org/jira/browse/MNG-6704?redirect=false)** and commented

`new ChainedDependencyGraphTransformer` seems make no sense here.

```java
public static DefaultRepositorySystemSession newSession()
{
// ...
DependencyGraphTransformer transformer =
new ConflictResolver( new NearestVersionSelector(), new JavaScopeSelector(),
new SimpleOptionalitySelector(), new JavaScopeDeriver() );

// seems unnecessary
new ChainedDependencyGraphTransformer( transformer, new JavaDependencyContextRefiner() );

session.setDependencyGraphTransformer( transformer );
// ...
}
```

---

**Affects:** 3.6.1

0 votes, 5 watchers

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

May this be a bug? I assume that the `ChainedDependencyGraphTransformer` should go to `setDependencyGraphTransformer()`. [Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte), WDYT?

**Comment by jira-importer** [comment]
**[Robert Scholte](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=rfscholte)** commented

For me it would be the same guess. Maybe a blame or commitlog shows some more hints what is intended here.

**Comment by jira-importer** [comment]
**[guofei](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ief2009)** commented

The last modification time is Feb. 2013 by [bentmann](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=bentmann).

I have no idea about `JavaDependencyContextRefiner` for. Maybe unnecessary now ?

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

[guofei](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=ief2009), you want to fix that line and run all ITs and see what happens.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

I have pushed a branch. All ITs pass locally for me.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

ITs seem to pass. Unrelated ITs failing. Should be merge this?

**Comment by jira-importer** [comment]
**[Sylwester Lachiewicz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=slachiewicz)** commented

Our test case with same issue: [AbstractArtifactComponentTestCase.java#L360-L364](https://github.com/apache/maven/blob/master/maven-compat/src/test/java/org/apache/maven/artifact/AbstractArtifactComponentTestCase.java#L360-L364)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

[Sylwester Lachiewicz](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=slachiewicz),  an you push another patch on top of the current branch?

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Changed test case, pushed a branch. If all is fine, I will merge into 3.6.2.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [690841e9be62a93b468d48364b35608dd1b0c2a9](https://gitbox.apache.org/repos/asf?p=maven.git;a=commit;h=690841e9be62a93b468d48364b35608dd1b0c2a9).
